File: rust/converter/conv.rs
use std::io;

fn main() {
    loop {
        println!("Temperature Converter");
        println!("1. Celcius to Fahrenheit");
        println!("2. Fahrenheit to Celcius");
        println!("3. Exit");
        println!("4. Enter your choice: ");

        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read input");
        let choice: u32 = match input.trim().parse() {
            Ok(num) => num,
            Err(_) => {
                println!("Invalid input. Please enter a number");
                continue;
            }
        };

        match choice {
            1 => {
                println!("Enter Temperature in Celcius: ");
                let celcius = read_temperature();
                let fahrenheit = celcius_to_fahrenheit(celcius);
                println!("Temperature in Fahrenheit: {:.2}", fahrenheit);
            }
            2 => {
                println!("Enter Temperature in Fahrenheit: ");
                let fahrenheit = read_temperature();
                let celcius = fahrenheit_to_celcius(fahrenheit);
                println!("Temperature in Celcius: {:.2}", celcius);
            }
            3 => break,
            _ => println!("Invalid choice. Please enter a number between 1 and 3"),
        }
    }
}

fn read_temperature() -> f64 {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read input");
    input.trim().parse().expect("Invalid temperature")
}

fn celcius_to_fahrenheit(celcius: f64) -> f64 {
    celcius * 9.0 / 5.0 + 32.0
}

fn fahrenheit_to_celcius(fahrenheit: f64) -> f64 {
    (fahrenheit - 32.0) * 5.0 / 9.0
}

File: go/conv.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func celciusToFahrenheit(celcius float64) float64 {
	return (celcius * 5 / 9) + 32
}

func fahrenheitToCelcius(fahrenheit float64) float64 {
	return (fahrenheit - 32) * 5 / 9
}

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Println("Temperatur Converter")
	fmt.Println("1. Celcius to Fahrenheit")
	fmt.Println("2. Fahrenheit to Celcius")
	fmt.Print("Enter your choice (1 or 2): ")
	choiceStr, _ := reader.ReadString('\n')
	choice, err := strconv.Atoi(choiceStr[:len(choiceStr)-1])
	if err != nil || (choice != 1 && choice != 2) {
		fmt.Println("Invalid choice. Please enter 1 or 2.")
		return
		
	}

	var inputUnit, outputUnit string
	var convert func(float64) float64

	if choice == 1 {
		inputUnit = "Celcius"
		outputUnit = "Fahrenheit"
		convert = celciusToFahrenheit
	} else {
		inputUnit = "Fahrenheit"
		outputUnit = "Celcius"
		convert = fahrenheitToCelcius
	}

	fmt.Printf("Enter the temperature in %s: ", inputUnit)
	tempStr, _ := reader.ReadString('\n')
	temp, err := strconv.ParseFloat(tempStr[:len(tempStr)-1], 64)
	if err != nil {
		fmt.Println("Invalid input. Please enter a valid temperature.")
		return
		
	}

	result := convert(temp)
	fmt.Printf("The temperature %.2f %s is %.2f %s\n", temp, inputUnit, result, outputUnit)
	
	
	
	
	
}
